File: backend/inventory-app/src/main/java/com/inventory/adapters/web/dto/category/CategoryResponse.java
package com.inventory.adapters.web.dto.category;

import java.time.Instant;
import java.util.UUID;

/**
 * DTO de respuesta para Categoría.
 */
public record CategoryResponse(
    UUID id,
    UUID parentId,
    String name,
    String path,
    int level,
    int sortOrder,
    boolean active,
    Instant createdAt,
    Instant updatedAt,
    int version
) {}

File: frontend/src/core/loading/validators/category-response.ts
import { z } from 'zod';
import type { CachedCategory } from '@/infrastructure/storage/db';

const nullableString = z.string().nullable().optional().default(null);

export const categoryResponseSchema: z.ZodSchema<CachedCategory> = z.object({
  id: z.string(),
  parentId: nullableString,
  name: z.string(),
  path: z.string(),
  level: z.coerce.number().int(),
  sortOrder: z.coerce.number().int(),
  active: z.boolean(),
  createdAt: z.string(),
  updatedAt: z.string(),
  cachedAt: z.coerce.number().optional(),
});
